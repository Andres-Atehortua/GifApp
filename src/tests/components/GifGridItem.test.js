import React from "react";
import GifGridItem from "components/GifGridItem";
import { shallow } from "enzyme";

describe("Pruebas al componente <GifGridItem />", () => {
  const title = "Soy el titulo";
  const url = "https://localhost/algo.jpg";
  let wrapper = shallow(<GifGridItem title={title} url={url} />);
  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });
  test("should render correctly the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should have a p tag with the title", () => {
    const pText = wrapper.find("p").text().trim();

    expect(pText).toBe(title);
  });

  test("should have an img with src and alt properties", () => {
    const { src, alt } = wrapper.find("img").props();

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should have class animate__backInRight in the div className", () => {
    const div = wrapper.find("div").hasClass("animate__backInRight");
    expect(div).toBeTruthy();
  });
});
