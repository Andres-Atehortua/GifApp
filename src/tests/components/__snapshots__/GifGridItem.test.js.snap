// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGridItem /> should render correctly the component 1`] = `
<div
  className="card animate__animated animate__backInRight"
>
  <img
    alt="Soy el titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Soy el titulo
  </p>
</div>
`;
